File: ruby/hamming/hamming.rb
module BookKeeping
    VERSION = 3 # Where the version number matches the one in the test.
end

class Hamming

  def self.compute(string1,string2)
  
    if string1.length != string2.length
        raise ArgumentError.new("Strings must be of equal length")
    else
      i = 0
      count = 0

      while i < string1.length do

        if string1[i] != string2[i]
          count += 1
        end
        i += 1
      end
      return count
    end
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(string1, string2) {

  var count = 0;

  string1Array = string1.split("");
  string2Array = string2.split("");

  if (string1Array.length === string2Array.length){
    for (var i = 0; i < string1Array.length; i++) {

      if (string1Array[i] != string2Array[i]) {
        count ++;
      }
      
    }

    return count;
  }

  else {
    throw  new Error('DNA strands must be of equal length.')
  }

}

module.exports = Hamming;


